### PR TITLE
add napari plugin

### PR DESCRIPTION
**Pull request by kevinyamauchi**
This adds the initial sample data plugin for the hiv dataset.

I put the sample data function in the same file as the dataset. I think that makes sense so that everything related to a dataset is in a single file.

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
Loading sample data from a cached image!


https://user-images.githubusercontent.com/1120672/206909850-dd6e9260-695b-4b6c-bf52-bd76ce985ae7.mov

File: src/cryo_et_sample_data/_hiv.py [hunk 1]
```diff
@@ -20,3 +22,24 @@
 }
 
 hiv = DataSet.from_dict(hiv_config)
+
+
+def _hiv_sample_tomogram() -> List["LayerDataTuple"]:  # noqa: F821
```
Comment (kevinyamauchi) [inline src/cryo_et_sample_data/_hiv.py]: @alisterburt , do you know how to make flake8 accept the forward reference? I'd rather not depend on napari just for this type annotation. This `noqa` works - just wondering if there's a better way.

Comment (alisterburt) [reply src/cryo_et_sample_data/_hiv.py]: I *think* this should work at the top of the file

```python
from __future__ import annotations
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from napari.types import LayerDataTuple
```

**Review by alisterburt — APPROVED** [review]
oh I loooooove this - my only minor thing would be that ideally these contributions would end up in nested menus, one per dataset and a menu for the whole plugin. Do you know if that's possible? I had a quick look at [the docs](https://napari.org/stable/plugins/contributions.html) but couldn't see it

cc @dragadoncila - is it possible to have nested menus of sample data contributions?

**Comment by brisvag** [comment]
This is great! One thing I'm not sure about: as you know I wanted to be able to use these through https://github.com/brisvag/blik for easy testing/showcase. Is there a way to use a specific plugin to read the sample data?

**Comment by alisterburt** [comment]
I'm not sure what you mean @brisvag - having this plugin installed will make the sample data available from the sample data menu, if you add it as a dep you should be good?

**Comment by brisvag** [comment]
If I understand correctly, the way sample data works is that it simply returns a pre-made napari layer, so it won't be read by a specific reader. Or am I misunderstanding? If that's the case, it won't be compatible with blik, cause the scale won't be set correctly, and it won't have the metadata necessary to know which `experiment_id` it belongs to.

**Comment by alisterburt** [comment]
ah! there are two layers to this package - non napari and napari, you can depend on this, get the data saved on disk and access it directly from the hiv instance of the dataset... check the example for how :)

**Comment by alisterburt** [comment]
forward ref fixed and added a tqdm dep which is required for running the example - will merge when tests pass

**Comment by brisvag** [comment]
> ah! there are two layers to this package - non napari and napari, you can depend on this, get the data saved on disk and access it directly from the hiv instance of the dataset... check the example for how :)

Awesome! However, installing the package will show the dataset in napari sample data menu, so all I can do from blik is "double it" and have another menu entry :/

**Comment by alisterburt** [comment]
Yeah not really sure what to suggest there...

**Comment by brisvag** [comment]
Anyways, it works for now (and it specifies which plugin it comes from, so that's at least distinguishable). Only thing is that I need to use `_get_data()` (so I can just pass the path to the cryohub reader), which is private here, so it would be nice to expose that.

**Pull request merged**